deepchem/models/tensorgraph/layers.py

```diff
@@ -567,6 +589,32 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     return out_tensor
 
 
+class Conditional(Layer):
+  """
+  A Conditional Layer wraps tf.Cond. 
```

Comment:
How about describing what it does rather than asking the reader to look up Tensorflow documentation to find out?

Reply:
I can add a bit more, but the actual semantics of tf.Cond is extremely nuanced that I'd almost prefer the reader to read the official documentation. There's tiny bits like:

Note that the conditional execution applies only to the operations defined in true_fn and false_fn. Consider the following simple program:

z = tf.multiply(a, b)
result = tf.cond(x < y, lambda: tf.add(x, z), lambda: tf.square(y))
If x < y, the tf.add operation will be executed and tf.square operation will not be executed. Since z is needed for at least one branch of the cond, the tf.multiply operation is always executed, unconditionally. Although this behavior is consistent with the dataflow model of TensorFlow, it has occasionally surprised some users who expected a lazier semantics.

Reply:
Then say that! :)  "Outputs `true_value` if the condition evaluates to true, or `false_value` if the condition evaluates to false.  There are many subtleties to the behavior of this layer, and it sometimes behaves in unexpected ways.  See the documentation for `tensorflow.cond()` for details."